packages/vite/src/node/optimizer/index.ts

```diff
@@ -1042,14 +1042,36 @@ function isSingleDefaultExport(exports: readonly string[]) {
 }
 
 const lockfileFormats = [
-  'package-lock.json',
-  'yarn.lock',
-  'pnpm-lock.yaml',
-  'bun.lockb'
+  { name: 'package-lock.json', patchesDirs: ['patches'] }, // Default of https://github.com/ds300/patch-package
+  { name: 'yarn.lock', patchesDirs: ['patches', '.yarn/patches'] }, // .yarn/patches for v2+
```

Comment:
This can be changed by the user but either way it's part of the lockfile so this shouldn't be necessary.

Reply:
Thanks for the report, I didn't notice while I was testing. I will send a PR to fix this before the release